utils/include/detray/simulation/event_generator/random_track_generator.hpp

```diff
@@ -16,47 +16,77 @@
 
 // System include(s)
 #include <algorithm>
-#include <cmath>
+#include <array>
 #include <random>
 
 namespace detray {
 
-/// Wrapper for random number generatrion for the @c random_track_generator
+/// Wrapper for CPU random number generatrion for the @c random_track_generator
 template <typename scalar_t = scalar,
           typename distribution_t = std::uniform_real_distribution<scalar_t>,
-          typename generator_t = std::random_device,
           typename engine_t = std::mt19937_64>
 struct random_numbers {
 
-    random_numbers(random_numbers&& other) : engine(std::move(other.engine)) {}
+    std::seed_seq m_seeds;
+    engine_t m_engine;
 
-    generator_t gen;
-    engine_t engine;
+    /// Random seed for increased entropy. Different random numbers each time
+    DETRAY_HOST
+    random_numbers() : m_seeds{get_random_seeds()}, m_engine{m_seeds} {}
```

Comment:
We can consider removing this constructor or set as `m_seeds{0u}`

Reply:
Zero is not a good seed and using a fixed value introduces no entropy at all

Reply:
Hmm what defines a goodness of seed?

Reply:
I am clearly not an expert, I just want to minimize bias, repetitions etc (maybe not so important to us?)  I would naively expect, the more "randomness" in the seed bits the better the statistical properties of the random numbers (but this might also be wrong?), so I would be suspicious about setting all seed bits to 0. Again, I might be totally wrong here, but I would rather like to be on the side of caution, unless this clashes with your work?

Reply:
> so I would be suspicious about setting all seed bits to 0. 

I don't get this. A seed is configurable so it is not going to be always zero as long as the user set the number carefully.

If the random seed sequence is not reproducible for every run time by default, it is just going to increase confusion to people who gets different output every time without knowing what's going on.

But I will always set the seed anyway for reproducibility so not my problem.
We can go with that for sure